examples/platform/silabs/SoftwareFaultReports.cpp

```diff
@@ -83,6 +83,19 @@ void OnSoftwareFaultEventHandler(const char * faultRecordString)
 } // namespace DeviceLayer
 } // namespace chip
 
+// This method is already implemented in the Zigbee stack and is required by the Zigbee
+#ifndef SL_CATALOG_ZIGBEE_STACK_COMMON_PRESENT
+extern "C" void halInternalAssertFailed(const char * filename, int linenumber)
+{
+    char faultMessage[kMaxFaultStringLen] = { 0 };
+    snprintf(faultMessage, sizeof faultMessage, "Assert failed: %s:%d", filename, linenumber);
+#if SILABS_LOG_ENABLED
```

Comment:
Shouldn't this be move on top of the faultMessage declaration ? I mean if the logs are not enabled, the buffer declaration and assignment seems pointless...